src/jdk.compiler/share/classes/com/sun/tools/javac/tree/TreeInfo.java

```diff
@@ -344,11 +344,13 @@ public void visitExec(JCExpressionStatement stat) {
         @Override
         public void visitClassDef(JCClassDecl tree) {
             // don't descend any further
+            result = tree;
```

Comment:
This was an issue in the current code: this is a tree translator (not just a tree scanner) and a tree translator has to side-effect the `result` variable, otherwise clients can see `null` popping up in strange places (which is what started happening when I moved `LambdaToMethod` around).